plugins/modules/vmware_guest.py

```diff
@@ -2433,12 +2429,17 @@ def set_disk_parameters(self, disk_spec, expected_disk_spec, reconfigure=False):
             disk_spec.device.capacityInKB = kb
             disk_modified = True
 
+        # set datastore per disk on creation only.
+        if expected_disk_spec.get('datastore'):
+            ds = find_obj(self.content, [vim.Datastore], expected_disk_spec.get('datastore'))
```

Comment:
Please check if `ds` is None or VMware object before proceeding.

Reply:
@Tinashecj please adjust the code like @Akasurde said. It looks like this is actually causing the `"msg": "Failed to create a virtual machine : Invalid configuration for device '1'."` error that we see in the CI.